Feature/PT-1/main-page

dungeon-slave opened the pull request:
Create main_menu_screen widget
Create dish_item widget
Create tab_bar widget
End project setup
Add DI
Add all layers
Add all necessary widgets
Implement MainMenu feature
Add themes

https://github.com/dungeon-slave/food_app/assets/70040363/4af4b4ea-05e8-462f-8b27-32fd017cb8dd

dungeon-slave committed:
[PT-1] project-setup create main packages and plug for main_menu_screen

dungeon-slave committed:
[PT-1] project-setup add firebase to core

dungeon-slave committed:
[PT-1] project-setup add flutter bloc to core

dungeon-slave committed:
configure firebase

dungeon-slave committed:
[PT-1] project-setup configure firebase

dungeon-slave committed:
Merge branch 'feature/PT-1/project-setup' of https://github.com/dungeon-slave/market_client into feature/PT-1/project-setup

dungeon-slave committed:
[PT-1] project-setup fix firebase location

dungeon-slave committed:
[PT-1] main-page
Add main_menu widget
Add item widget

dungeon-slave committed:
[PT-1] main-page
Create main_menu_screen widget
Create dish_item widget
Create tab_bar widget

dungeon-slave committed:
[PT-1] main-page
End project setup
Add DI
Add all layers
Add all necessary widgets
Implement MainMenu feature

dungeon-slave committed:
[PT-1] main-page
Small TYPO fix

dungeon-slave committed:
[PT-1] main-page
TYPO fix

dungeon-slave committed:
[PT-1] main-page
Add themes

dungeon-slave committed:
[PT-1] main-page
Remove unnecessary
TYPO fixes

Alexey-Lebedzeu commented on core/lib/di/data_di.dart:
перениси дата ди в дату , не делай лишнюю зависимость в коре

Alexey-Lebedzeu commented on core/pubspec.yaml:
```diff
@@ -10,45 +10,17 @@ environment:
+  data:
+    path: ../data
+  domain:
+    path: ../domain
```
и кор по идее не должен зависеть от даты и домена

Alexey-Lebedzeu commented on core_ui/lib/src/theme/colors.dart:
```diff
@@ -0,0 +1,7 @@
+import 'package:flutter/material.dart';
+
+class MyColors {
```
naming

Alexey-Lebedzeu commented on core_ui/lib/src/theme/dimens.dart:
```diff
@@ -0,0 +1,9 @@
+class Dimens {
+  static const double dishesPadding = 10;
```
сделай что-то вроде `  static const double BORDER_RADIUS_48 = 48.0;
  static const double BORDER_RADIUS_100 = 100.0;

  static const double PADDING_2 = 2.0;
`

тк значения переиспользуются,

Alexey-Lebedzeu commented on core_ui/lib/src/theme/fonts.dart:
```diff
@@ -0,0 +1,23 @@
+import '../../core_ui.dart';
+
+class Fonts {
+  static const dishNameFont = TextStyle(
```
сделай дефолты шрифтов типо `
  static TextStyle bold22 = TextStyle(
    fontFamily: GoogleFonts.roboto().fontFamily,
    fontWeight: FontWeight.w600,
    fontSize: 22,
  );
` тк они опять же могут использоватся много где одинаковыми

Alexey-Lebedzeu commented on data/lib/data.dart:
```diff
@@ -1,7 +1 @@
-  /// Returns [value] plus 1.
-  int addOne(int value) => value + 1;
-}
+library;
```
просто библиотека?? а где название ?

Alexey-Lebedzeu commented on data/lib/providers/firebase_provider.dart:
```diff
@@ -0,0 +1,45 @@
+
+class FirebaseProvider {
+
+  late final List<DishEntity> dishEntities = [];
```
очень плохая идея.. хранить инфу прямо в провайдере. делай или стор локальный или храни уже загруженные в блоке, но никак не в репе или провайдере

Alexey-Lebedzeu commented on data/lib/providers/firebase_provider.dart:
```diff
@@ -0,0 +1,45 @@
+
+    List<DishEntity> filteredDishes = [];
+
+    for (var dish in dishEntities) {
```
var ?

Alexey-Lebedzeu commented on data/lib/providers/firebase_provider.dart:
```diff
@@ -0,0 +1,45 @@
+
+    final fbMenu = (await FirebaseFirestore
+                      .instance
+                      .collection('menu')
```
константы коллекций вынеси в кор в файл констант

Alexey-Lebedzeu commented on data/lib/providers/firebase_provider.dart:
```diff
@@ -0,0 +1,45 @@
+                      .toList();
+
+    for (var fbDish in fbMenu) {
+      dishEntities.add(DishEntity(
```
напиши методы для энтити fromJson . или используй сериализаторы

Alexey-Lebedzeu replied:
лучше конечно второй вариант

Alexey-Lebedzeu commented on data/lib/providers/firebase_provider.dart:
```diff
@@ -0,0 +1,45 @@
+      ));
+    }
+
+    return;
```
что за пустой ретурн

dungeon-slave replied:
> что за пустой ретурн

Void же) Просто мне кажется так нагляднее где функция работу заканчивает. Или не надо в void-функциях return писать?

Alexey-Lebedzeu commented on data/lib/providers/firebase_provider.dart:
```diff
@@ -0,0 +1,45 @@
+
+  Future<void> _getDishes() async {
+
+    final fbMenu = (await FirebaseFirestore
```
где тип ?

Alexey-Lebedzeu commented on data/lib/repositories/dishes_repository_impl.dart:
```diff
@@ -0,0 +1,18 @@
+import 'package:domain/models/dish_model.dart';
+import 'package:domain/repositories/dishes_repository_interface.dart';
+
+class DishesRepository implements IDishesRepository {
```
что с нэймингом ?

dungeon-slave replied:
Ты про IDishesRepository? Объясняю смысл мува: я смотрел тестовый что вы кидали и там интерфейсы называются как обычные классы, а у реализации в конце Impl добавляется. Я подумал, что удобнее и нагляднее будет в интерфейсах спереди добавлять букву I, а остальные классы оставить как есть. Тк так ты сразу понимаешь что перед тобой интерфейс находится, а в случае с Impl мне надо всё название класса до конца дочитать, что трудозатратнее. Ну если так делать нельзя, то я буду писать Impl в конце реализаций

dungeon-slave replied:
Просто, мне кажется, это из разряда тех вещей как вы с командой договоритесь. Я вот с собой так говорился. Хорошая практика, в Си Шарпе по дефолту используется, в тайпскрипте встречал

Alexey-Lebedzeu commented on domain/lib/repositories/dishes_repository_interface.dart:
```diff
@@ -0,0 +1,5 @@
+import '../models/dish_model.dart';
+
+abstract class IDishesRepository {
```
что с неймингом ?

Alexey-Lebedzeu commented on domain/lib/usecase/fetch_dishes_usecase.dart:
```diff
@@ -0,0 +1,14 @@
+import 'package:domain/models/dish_model.dart';
+import 'package:domain/usecase/usecase.dart';
+
+class FetchDishesUsecase implements IUseCase<String, List<DishModel>> {
```
IUseCase????

Alexey-Lebedzeu commented on domain/lib/usecase/usecase.dart:
```diff
@@ -0,0 +1,3 @@
+abstract class IUseCase<Input, Output> {
```
naming

Alexey-Lebedzeu commented on domain/pubspec.yaml:
```diff
@@ -10,45 +10,10 @@ environment:
   flutter:
     sdk: flutter
+  core:
+    path: ../core
```
зачем?

Alexey-Lebedzeu commented on features/main_menu/lib/src/ui/dish_item.dart:
```diff
@@ -0,0 +1,78 @@
+  final String _imageRef;
+  final String _price;
+
+  const DishItem(this._name, this._imageRef, this._price, {super.key});
```
if you have more then 1 parameters make they named parameter

Alexey-Lebedzeu commented on features/main_menu/lib/src/ui/dish_item.dart:
```diff
@@ -0,0 +1,78 @@
+  Widget build(BuildContext context) {
+    return Container(
+      margin: const EdgeInsets.only(
+          top: Dimens.vertCardMargin, bottom: Dimens.vertCardMargin),
```
add ,

Alexey-Lebedzeu commented on features/main_menu/lib/src/ui/dish_item.dart:
```diff
@@ -0,0 +1,78 @@
+      margin: const EdgeInsets.only(
+          top: Dimens.vertCardMargin, bottom: Dimens.vertCardMargin),
+      padding: const EdgeInsets.only(
+          top: Dimens.vertCardPadding, bottom: Dimens.vertCardPadding),
```
,

Alexey-Lebedzeu commented on features/main_menu/lib/src/ui/dish_item.dart:
```diff
@@ -0,0 +1,78 @@
+        borderRadius: BorderRadius.all(Radius.circular(Dimens.cardRadius)),
+      ),
+      child: Column(
+        children: [
```
type

Alexey-Lebedzeu commented on features/main_menu/lib/src/ui/dish_item.dart:
```diff
@@ -0,0 +1,78 @@
+            widget._name,
+            style: Fonts.dishNameFont,
+          ),
+          Image.network(
```
create one image for whole app like AppImage

Alexey-Lebedzeu commented on features/main_menu/lib/src/ui/dish_item.dart:
```diff
@@ -0,0 +1,78 @@
+          ),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+            children: [
```
type

Alexey-Lebedzeu commented on features/main_menu/lib/src/ui/tab_bar.dart:
```diff
@@ -0,0 +1,82 @@
+                    isScrollable: true,
+                    onTap: (value) {
+                      BlocProvider.of<MenuBloc>(context)
+                          .add(ChangeTypeEvent(dishesTypes[value]!));
```
,

Alexey-Lebedzeu commented on features/main_menu/lib/src/ui/tab_bar.dart:
```diff
@@ -0,0 +1,82 @@
+    return BlocProvider<MenuBloc>(
+      create: (BuildContext context) =>
+          MenuBloc(appLocator.get<FetchDishesUsecase>())
+            ..add(ChangeTypeEvent(dishesTypes[0]!)),
```
,

Alexey-Lebedzeu commented on features/main_menu/lib/src/ui/tab_bar.dart:
```diff
@@ -0,0 +1,82 @@
+    0: 'Burgers',
+    1: 'Pizza',
+    2: 'Drinks'
+  };
```
move to state

dungeon-slave committed:
[PT-1] main-page
add freezed package
dependencies changes
add AppImage widget
core_ui typo fixes
add AppConstants
add models caching in bloc
rework firebase provider

dungeon-slave committed:
[PT-1] main-page
change dependencies
rework bloc
add dynamic dishes types
rework menu widget
add swipes between dishes types
remove unnecessary

dungeon-slave committed:
[PT-1] main-page
format files
